src/workerd/api/pyodide/pyodide.c++

```diff
@@ -56,6 +58,32 @@ int PyodideMetadataReader::read(jsg::Lock& js, int index, int offset, kj::Array<
   return toCopy;
 }
 
+int PyodideMetadataReader::readMemorySnapshot(int offset, kj::Array<kj::byte> buf) {
+  int snapshotSize = memorySnapshot.size();
+  if (offset >= snapshotSize || offset < 0) {
+    return 0;
+  }
+  int toCopy = buf.size();
+  if (snapshotSize - offset < toCopy) {
+    toCopy = snapshotSize - offset;
+  }
+  memcpy(buf.begin(), &memorySnapshot[0] + offset, toCopy);
```

Comment:
Hmm, I'm concerned this may be unsafe.

@jasnell the `buf` here comes from JS, can you verify that doing a `memcpy` like this is safe?

Reply:
We already have an almost identical copy of this code elsewhere in this same file which James reviewed =)